docs/source/faq.rst

```diff
@@ -43,13 +43,15 @@ in tools like ``ps`` and ``top``. This helps for distinguishing the master
 process as well as between masters when running more than one app on a single
 machine. See the proc_name_ setting for more information.
 
-Why is there no HTTP Keep-Alive?
---------------------------------
+Why is there no HTTP Keep-Alive header?
+---------------------------------------
 
-The default Sync workers are designed to run behind Nginx which only uses
-HTTP/1.0 with its upstream servers. If you want to deploy Gunicorn to
-handle unbuffered requests (ie, serving requests directly from the internet)
-you should use one of the async workers.
+You need to enable Keep-Alive support in both gunicorn and your load balancer.
+To enable Keep-Alive support in gunicorn you must use an async worker class,
+which is any of the worker classes that aren't the default sync class. To
+enable Keep-Alive with nginx you must enable its HTTP 1.1 support. Add
+``proxy_http_version 1.1;`` and ``proxy_set_header Connection "";`` to your
```

Comment:
Is it definitely good to add `proxy_set_header Connection ""`? Downstream clients should not be sending `Connection: close` if they want to make a keep-alive request. If the advice is just about how to get keep-alive between a reverse proxy and Gunicorn, then this makes sense, but it also prohibits `Connection: upgrade` for websockets requests, doesn't it?

Reply:
Possibly. This is getting out of my familiarity with HTTP, I don't know what a good suggestion would be here.

Reply:
I think let's remove the `proxy_set_header` and I will merge this.